libr/debug/p/native/linux/linux_debug.c

```diff
@@ -428,12 +433,12 @@ static bool linux_attach_single_pid(RDebug *dbg, int ptid) {
 			perror ("ptrace (PT_ATTACH)");
 			return false;
 		}
-
-		if (!linux_set_options (dbg, ptid)) {
-			return false;
-		}
```

Comment:
You won't be able to set thread options this way. I would suggest to restore this line for thread option setting and to add the following code specifically for the main, potentially TRACEME'd, tid

    else if (ptid == dbg->main_pid && !dbg->options_set) {
        if (!linux_set_options (dbg, ptid)) {
            return false;
        }
    }

Reply:
Mmm what about we just always call linux_set_options no matter what? Is there any issue with that?

Reply:
I looked into it and there doesn't seem to be a side effect to constantly changing thread options, that would be the easiest solution.

Reply:
ok perfect I will go with the easiest solution for now, as it doesn't have side effects and it worked/works.